e2e/tests/node_not_ready/node_disconnection.robot

```diff
@@ -0,0 +1,150 @@
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds    ${retry_interval} seconds    Volume state should be healthy
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds     ${retry_interval} seconds    Engine state should be running
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds    ${retry_interval} seconds    Replica on the node ${node_number_with_replica}[0] state should be running
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds    ${retry_interval} seconds    Replica on the node ${node_number_with_replica}[1] state should be running
+    [Teardown]    Run Keywords   Settings auto-salvage is true
+    ...           AND            cleanup_resources
+
+6-RWX 2 replicas abnormal non attached no replica node
+    [Documentation]    Case 2-1
+    [Tags]    NodeNetworkPartition    RWX    2Replicas    AttachReplicaSameNode    AbnormalNoReplicaNode    AbnormalNoAttachedNode  
+    Given Create ${volume_size_gb} GB RWX volume with 2 replicas
+    And Settings auto-salvage is false
+    And Get node index with replica
+    And Get node index without replica
+    And Attach volume to the node ${node_number_without_replica}[0]
+    When Interrupt network the node ${node_number_without_replica}[0] for 100 seconds
+    Then Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds    ${retry_interval} seconds    Node ${node_number_without_replica}[0] state should be Ready
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds    ${retry_interval} seconds    Volume state should be healthy
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds     ${retry_interval} seconds    Engine state should be running
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds    ${retry_interval} seconds    Replica on the node ${node_number_with_replica}[0] state should be running
+    And Run keyword And Continue On Failure    Wait Until Keyword Succeeds    ${retry_time_out} seconds    ${retry_interval} seconds    Replica on the node ${node_number_with_replica}[1] state should be running
+    [Teardown]    Run Keywords   Settings auto-salvage is true
+    ...           AND            cleanup_resources
+
+7-RWO 3 replicas abnormal non attached no replica node
+    [Documentation]    Case 2-2
+    [Tags]    NodeNetworkPartition    RWO    3Replicas    AttachReplicaSameNode    AbnormalReplicaNode    AbnormalAttachedNode   
+    Given Create ${volume_size_gb} GB RWO volume with 3 replicas
```

Comment:
`Given`, `And`, `When`, `Then` are not mandatory. We can omit them if it's tedious. Like `Given Create ...` is a little weird.

Reply:
We need to keep these terms which are specific for BDD. 

It will be helpful to understand the context of the test case.

Reply:
This is an example. Just for reference.
https://github.com/longhorn/longhorn-tests/blob/f8e716273ab243ea07b7c51e646566b5b297e740/e2e/tests/replica_resiliency_test.robot#L15

Reply:
I probably know what u are concerned. Let's discuss more in the meeting to settle the test case convention.

Reply:
In my opinion, I prefer sticking with the BDD style because it makes the test case easier to understand with clearly defined `preparation` (Given), `test` (When), and `expectation` (Then) phases.

Regarding `Create ${volume_size_gb} GB RWO volume with 3 replicas`, I believe it's better to break it up into multiple keywords. This approach helps maintain readability, especially when adding more configurations in the future. For example:
```
Given Prepare ${size} GB Volume ${vol_name}
And Volume ${volume_name} is prepared with ${num_of_replicas} replicas
And Volume ${volume_name} is prepared with ${volume_type} access mode
And Volume ${volume_name} created
```

Reply:
I think it's possible that we can put `Create X GB RWO volume with 3 replicas` in *.robot (test case definition).

And put something like
```
Given Prepare ${size} GB Volume ${volume_name}
And Volume ${volume_name} is prepared with ${number_of_replicas} replicas
And Volume ${volume_name} is prepared with ${volume_type} access mode
And Volume ${volume_name} is created
```
in *.resource (custom keyword implementation).

Reply:
I will try to process the volume preparation to separate as several steps in `.resource` file and do my best to make it more maintainable and extensible.

Reply:
I've separate the volume creation into multiple steps: prepare volume object, update the replica count, update the volume type. And the volume could be created normally.

![image](https://github.com/longhorn/longhorn-tests/assets/17548901/c05cba51-50b7-4113-b2df-d46ba4b2e06c)